internal/engine/upper.go

```diff
@@ -34,16 +35,22 @@ type Upper struct {
 	b            BuildAndDeployer
 	watcherMaker watcherMaker
 	timerMaker   timerMaker
+	k8s          k8s.Client
 }
 
 type watcherMaker func() (watch.Notify, error)
 type timerMaker func(d time.Duration) <-chan time.Time
 
-func NewUpper(ctx context.Context, b BuildAndDeployer) (Upper, error) {
+func NewUpper(ctx context.Context, b BuildAndDeployer, k8s k8s.Client) (Upper, error) {
```

Comment:
while you're in this code, is it easy to remove the spurious `error` return value?

Reply:
done!